plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergV2.java

```diff
@@ -229,6 +231,29 @@ public void testV2TableWithEqualityDeleteWhenColumnIsNested()
         assertQuery("SELECT array_column[1], map_column[1], row_column.x FROM " + tableName, "SELECT 1, 2, 1 FROM nation WHERE regionkey != 1");
     }
 
+    @Test
+    public void testParquetGeneratedByPyArrow()
+    {
+        String tableName = "test_parquet_by_pyarrow" + randomNameSuffix();
```

Comment:
Do we need pyarrow file specifically ?
My guess is that we could write file through hive connector and migrate table to iceberg to get similar situation

Reply:
I couldn't reproduce the issue with Hive connector. We already have such a test `TestIcebergMigrateProcedure#testMigrateTableWithComplexType`, but it didn't catch the regression.

Reply:
I looked closer at it, we're able to generate ids from `nameMapping` for the migrated tables, that's why there is no issue there.